### PR TITLE
Write fluxes

### PR DESCRIPTION
**Pull request by asjeb**
merge write_fluxes into main 
writes fluxes : 
            ["pn", "en", "pr", "perc", "lexc", "prr", "prd", "qr", "qd"],  # % gr4
            ["pn", "en", "pr", "perc", "lexc", "prr", "prd", "qr", "qd"],  # % gr5
            ["ei", "pn", "en", "pr", "perc", "prr", "qr"],  # % grd
            ["ei", "pn", "en", "pr", "perc", "prr", "prd", "qr", "qd"],  # % loieau
            ["pn", "en", "qr", "qb"],  # % vic3l

writes states : 
            ["hi", "hp", "ht"],  # % gr4
            ["hi", "hp", "ht"],  # % gr5
            ["hp", "ht"],  # % grd
            ["ha", "hc"],  # % loieau
            ["hcl", "husl", "hmsl", "hbsl"],  # % vic3l

writes stats :
           mean, var, minimum, maximum, median
when run_forward is worked with activated return_options

### CODE REVIEW

File: smash/fcore/operator/md_gr_operator.f90 [hunk 1]
```diff
@@ -277,6 +304,30 @@ subroutine gr5_time_step(setup, mesh, input_data, options, time_step, ac_mlt, ac
...
+                end if
+                if (returns%pn_flag) returns%pn(row, col, time_step) = pn
+                if (returns%en_flag) returns%en(row, col, time_step) = en
+                if (returns%pr_flag) returns%pr(row, col, time_step) = pr
+                if (returns%perc_flag) returns%perc(row, col, time_step) = perc
+                if (returns%lexc_flag) returns%lexc(row, col, time_step) = l
+                if (returns%prr_flag) returns%prr(row, col, time_step) = prr
+                if (returns%prd_flag) returns%prd(row, col, time_step) = prd
+                if (returns%qr_flag) returns%qr(row, col, time_step) = qr
+                if (returns%qd_flag) returns%qd(row, col, time_step) = qd
```
Comment (pag13) [inline smash/fcore/operator/md_gr_operator.f90:328]: ok for this first version. flux maps getter implemented?
Wouldn't this be better to use sthg like returns%internal_fluxes_maps(:,:,:,:) 
with 1..N fluxes which names is defined for each model structure (as for params or for stats%internal_fluxes above) in _constants.py?

File: smash/fcore/operator/md_vic3l_operator.f90 [hunk 2]
```diff
@@ -229,6 +244,19 @@ subroutine vic3l_time_step(setup, mesh, input_data, options, time_step, ac_mlt,
...
+                    returns%stats%internal_fluxes(row, col, 1) = pn
+                    returns%stats%internal_fluxes(row, col, 2) = en
+                    returns%stats%internal_fluxes(row, col, 3) = qr
+                    returns%stats%internal_fluxes(row, col, 4) = qb
+                end if
+                if (returns%pn_flag) returns%pn(row, col, time_step) = pn
+                if (returns%en_flag) returns%en(row, col, time_step) = en
+                if (returns%qr_flag) returns%qr(row, col, time_step) = qr
+                if (returns%qb_flag) returns%qb(row, col, time_step) = qb
+                !$AD end-exclude
```
Comment (pag13) [inline smash/fcore/operator/md_vic3l_operator.f90:259]: vertical fluxes between production store layers are missing

**Review by pag13 — COMMENTED** [review]
Nice job! 
This is only comments from a first reading of the code, without testing in details on a case.

File: smash/_constant.py [hunk 3]
```diff
@@ -867,6 +880,19 @@ def get_rr_states_from_structure(structure: str) -> list[str]:
...
 
+INTERNAL_FLUXES = dict(
+    zip(
+        HYDROLOGICAL_MODULE,
+        [
+            ["pn", "en", "pr", "perc", "lexc", "prr", "prd", "qr", "qd"],  # % gr4
+            ["pn", "en", "pr", "perc", "lexc", "prr", "prd", "qr", "qd"],  # % gr5
+            ["ei", "pn", "en", "pr", "perc", "prr", "qr"],  # % grd
+            ["ei", "pn", "en", "pr", "perc", "prr", "prd", "qr", "qd"],  # % loieau
+            ["pn", "en", "qr", "qb"],  # % vic3l
```
[the remark this answers is not in the document]
Comment (asjeb) [reply smash/_constant.py:891]: All drainage fluxes are on active cells...

**Review by pag13 — COMMENTED** [review]
Est que les nombreux commentaires et suggestions de FC ont été addressés.

Quid du mwd_stat ?

Quid de la resolution de pb de la diff auto ?

à investiguer ensemble et finaliser asap svp.

**Comment by pag13** [comment]
> Maintenant que l'on a les flux internes est ce que une classe stat python ne serait pas tout aussi adapté ? Ou bien qq chose à récupérer sur les gradients ?

Coucou, je ne suis pas sûr de bien comprendre, peux tu clarifier stp ?
Statistiques (sur dim spatiale et ou temporelle) de states/flux en Python - quid vs calcul de stat sur tableaux fortran vs memoire ?
Une fonction Python pour tracer des cartes de gradient serait très utile, Truyen a réalsié des scripts pr cela et publi WRR.

**Pull request closed without merging**